tensorflow/python/framework/dtypes.py

```diff
@@ -127,6 +127,11 @@ def as_numpy_dtype(self):
     """Returns a `numpy.dtype` based on this `DType`."""
     return _TF_TO_NP[self._type_enum]
 
+  def convert_to_numpy_obj(self, obj):
+    """Explicitly convert obj based on numpy type except for string type."""
+    numpy_dtype = self.as_numpy_dtype
+    return numpy_dtype(obj) if  numpy_dtype is not object else str(obj)
```

Comment:
Can you move this to be a private function in `session.py`? I'd prefer not to add it to the public API.

Reply:
but then.. we need to be careful not to add more `np.object` like types which... might not happen, but if.. then the session won't notice .. 
but anyway, i have addressed the comments